server/text_generation_server/utils/layers.py

```diff
@@ -606,24 +800,35 @@ def forward(self, x):
 
 
 class SpeculativeHead(nn.Module):
-    def __init__(self, lm_head, medusa):
+    def __init__(self, lm_head, speculator):
         super().__init__()
         self.head = lm_head
-        self.medusa = medusa
+        self.speculator = speculator
 
     @staticmethod
     def load(config, prefix: str, weights):
-        use_medusa = config.use_medusa
-        if use_medusa:
+        use_speculator = config.use_speculator
+        if use_speculator:
```

Comment:
I feel like this should switch from a bool to a true enum value directly (making the modifications more extensive).

`config.speculator = {None, "medusa", "MLPS"}` basically.

Reply:
I don't think it was ever a bool? it looked like it was a string path pointing to the speculator directory containing the config and weights?

```python
medusa_config = str(Path(use_medusa) / "config.json")
filename = str(Path(use_medusa) / "medusa_lm_head.safetensors")
```

We could keep make this an enum, or we can just keep as path and recover the class from the config? If this is possible with the current config.